Slim/Control/Queries.pm

```diff
@@ -4436,6 +4451,135 @@ sub yearsQuery {
 	$request->setStatusDone();
 }
 
+sub worksQuery {
+	my $request = shift;
+
+	# check this is the correct query.
+	if ($request->isNotQuery([['works']])) {
+		$request->setStatusBadDispatch();
+		return;
+	}
+
+	if (!Slim::Schema::hasLibrary()) {
+		$request->setStatusNotDispatchable();
+		return;
+	}
+
+	my $sqllog = main::DEBUGLOG && logger('database.sql');
+
+	# get our parameters
+	my $client        = $request->client();
+	my $index         = $request->getParam('_index');
+	my $quantity      = $request->getParam('_quantity');
+	my $work          = $request->getParam('work');
+	my $libraryID     = Slim::Music::VirtualLibraries->getRealId($request->getParam('library_id'));
+	my $hasAlbums     = $request->getParam('hasAlbums');
+	my $byComposer    = $request->getParam('byComposer');
```

Comment:
Please replace with the generic `sort`.

Reply:
I think this will go away, in favour of the works option under Composer (or Artist).